src/main/java/graphql/nadel/engine/Execution.java

```diff
@@ -47,12 +50,29 @@
     private final ExecutionHelper executionHelper = new ExecutionHelper();
     private final NadelExecutionStrategy nadelExecutionStrategy;
 
-    public Execution(List<Service> services, GraphQLSchema overallSchema, NadelInstrumentation instrumentation, IntrospectionRunner introspectionRunner, ServiceExecutionHooks serviceExecutionHooks) {
+    private NormalizedQueryFactory normalizedQueryFactory = new NormalizedQueryFactory();
+
+    public Execution(List<Service> services,
+                     GraphQLSchema overallSchema,
+                     NadelInstrumentation instrumentation,
+                     IntrospectionRunner introspectionRunner,
+                     ServiceExecutionHooks serviceExecutionHooks,
+                     Object userSuppliedContext) {
         this.services = services;
         this.overallSchema = overallSchema;
         this.instrumentation = instrumentation;
         this.introspectionRunner = introspectionRunner;
-        this.nadelExecutionStrategy = new NadelExecutionStrategy(services, createFieldsInfos(), overallSchema, instrumentation, serviceExecutionHooks);
+        FieldInfos fieldsInfos = createFieldsInfos();
+        if (userSuppliedContext instanceof BenchmarkContext) {
```

Comment:
Whats is this for?  Profilable code??

Reply:
yes: this colllects args for the benchmarks